ls_extension/main.go

```diff
@@ -0,0 +1,117 @@
+/*
```

Comment:
could we share code between main.go and this file?

Reply:
add entrypoint helpers, to at least share some snippets.